### DISCUSSION TITLE
[Core Protocol] how the new node fetcher sync/derivate the DA data if the blob expire?

### DISCUSSION THREAD

**Discussion by joohhnnn** [ethereum-optimism/developers#361] [Core Protocol] [answered]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

How the new node fetcher sync/derivate the DA data if the blob expire? we only talk about the derivation(from L1) case, let's ignore sync from other OP node. if we already have the solution, plz tag the main code here PLZ :)

**Comment by hash1go** [comment]
Hello!
Are you looking for a way to sync expired blob data to an L1 new node?

**Reply by joohhnnn** [reply]
not really. just want to clarify the code logic when the expired blob happened in the derivation(from L1) case

**Reply by hash1go** [reply]
So, are you looking for the specific code logic in the L1 go-ethereum codebase?

**Reply by joohhnnn** [reply]
No, it's not the logic in L1 GETH. Maybe I wasn't clear, let me try explaining it another way. Suppose I'm running my own non-sequencer node to verify data on the Optimism mainnet. I don't trust the data provided by other nodes and the sequencer, so I choose to directly obtain rollup data from L1 DA to maintain my node's data—in other words, to "sync." Now, if the blob data on L1 expires, how will the op-node handle it? Is there a configuration option set at the project's startup that routes all potentially expired L1 data, or how does it work?

**Reply by hash1go** [reply]
The op-node does not have the capability to regenerate its state using expired L1 DA data directly.
However, the state roots are recorded on L1, so the validity of any state regenerated using DA can be verified against the state roots. 

If needed, expired blob data still be stored by some Ethereum nodes, so you would need to request it from them.

**Reply by joohhnnn** [reply]
If that's the case, then the entire derivation might have issues. Additionally, relying on data provided by a first-layer network to replay the state of a second-layer seems impractical (unless additional data sources are provided).

**Reply by joohhnnn** [reply, marked as answer]
Never mind I think I found the solution ;)
<img width="1028" alt="image" src="https://github.com/ethereum-optimism/developers/assets/68833933/dafe3180-8df8-4f74-9408-a9719e89cc17">
